### PR TITLE
Upgrade iOS CI to CircleCI 2.0

### PR DESCRIPTION
**Pull request by brandonpage**
-  Run each libs tests in parallel on PRs, skipping libs that aren't modified.
-  Nightly builds of all tests for iOS 10 and iOS 11
-  Use Codecov.io (and coveralls.io just to test)
-  Move gem dependencies to gemfile for better reliability
-  Move all CI files to .circleci folder
-  Small amount of cleanup for fastlane file, added CircleCi flow while keeping Jenkins flow intact.
-  Add status badges

### CODE REVIEW

File: .circleci/DangerFile [hunk 1]
```diff
@@ -0,0 +1,69 @@
+gem 'plist'
```
Comment (brandonpage) [inline .circleci/DangerFile]: This file is not new, just moved to CI folder.

File: libs/SmartSync/SmartSync.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -3,7 +3,7 @@
 	archiveVersion = 1;
 	classes = {
 	};
-	objectVersion = 49;
```
Comment (brandonpage) [inline libs/SmartSync/SmartSync.xcodeproj/project.pbxproj]: I don't know where this came from but 48 is the latest.  This being 49 breaks some CI tools.

File: .circleci/Gemfile.lock [hunk 3]
```diff
@@ -0,0 +1,217 @@
+GEM
```
Comment (bhariharan) [inline .circleci/Gemfile.lock]: Shouldn't this be in `.gitignore`?

File: readme.md [hunk 4]
```diff
@@ -1,4 +1,6 @@
 [![Build Status](https://forcedotcom.github.io/SalesforceMobileSDK-TestResults/iOS-test-results/dev/latest/buildstatus.svg)](https://forcedotcom.github.io/SalesforceMobileSDK-TestResults/iOS-test-results/dev/latest/index.html)
```
Comment (bhariharan) [inline readme.md]: Should we remove the old line?

File: .circleci/Gemfile.lock (hunk 3 above)
Comment (brandonpage) [reply .circleci/Gemfile.lock]: Should be in source.

File: readme.md (hunk 4 above)
Comment (brandonpage) [reply readme.md]: Let's leave it for now.  We can drop it when we switch over for good.

Comment (bhariharan) [reply readme.md]: Sounds good.

**Comment by brandonpage** [comment]
If Codecov yells about coverage that's just because there isn't a good baseline on the dev branch.

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR, try to keep changes smaller if you can</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_PR-Check">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Classname | Name|
--- | ---|
ParentChildrenSyncTests | testReSyncWithUpdatedChildren|
ParentChildrenSyncTests | testReSyncWithUpdatedChildren|
ParentChildrenSyncTests | testReSyncWithUpdatedChildren|
ParentChildrenSyncTests | testReSyncWithUpdatedChildren|
ParentChildrenSyncTests | testReSyncWithUpdatedChildren|
ParentChildrenSyncTests | testReSyncWithUpdatedChildren|
ParentChildrenSyncTests | testReSyncWithUpdatedChildren|
ParentChildrenSyncTests | testReSyncWithUpdatedChildren|
ParentChildrenSyncTests | testReSyncWithUpdatedChildren|


# Tests results for SmartSync

<p align="right" data-meta="generated_by_SmartSync">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**